### PR TITLE
Job email and grouped results for stats

### LINKED ISSUE #44: Save notification email hash of job submission
- On posted job, generate a hash of the email (to make them anonymous) saved to the db.
- Allow search filter of job by email hash.

### PR DESCRIPTION
**Pull request by fmigneault**
- add `Job` email encrypted field 
- search jobs by email (hashed)
- group jobs by field categories

Resolves #44 

## todo
- complete #48

### CODE REVIEW

**Comment by fmigneault** [comment]
failing tests are related to #11 being implemented/fixed in https://github.com/crim-ca/weaver/tree/func-tests

File: weaver/store/mongodb.py [hunk 1]
```diff
@@ -477,6 +526,9 @@ def find_jobs(self,
         elif status:
             search_filters["status"] = status
 
+        if notification_email is not None:
+            search_filters["notification_email"] = notification_email
```
Comment (dbyrns) [inline weaver/store/mongodb.py]: I guess that we should search the encrypted mail and not the real mail?

Comment (fmigneault) [reply weaver/store/mongodb.py]: Done [here](https://github.com/crim-ca/weaver/pull/46/files/06a87735547afbf6a390868999e7d8f45557c4bf#diff-983083985a48e53e4a69189e6dbe2c53R207) and validated by [this test](https://github.com/crim-ca/weaver/pull/46/files/06a87735547afbf6a390868999e7d8f45557c4bf#diff-19c25203fab87905fceae3782f9a36ecR308)

Comment (fmigneault) [reply weaver/store/mongodb.py]: @tomLandry 
To filter by email, you have to specify the literal email to match against their hashed version in the db.
You can't search with the hashed version. 
Unless you have the secret & salt values to know how to pass from one-another, you can't find which hash corresponds to which email literal value, so identifies masking is preserved.

Furthermore, email values are not returned in responses. 
So you can only go by trial-error of literal email values to attempt matching jobs, and if you happen to match jobs this way, then you already have the email to create the request in the first place.

**Pull request merged**